### PR TITLE
GEODE-6720: Add FastLogger benchmarks

### PR DESCRIPTION
**Pull request by kirklund**
I'd like to merge these FastLogger benchmarks to develop:
* **FastLoggerDisabledLevelBenchmark** compares FastLogger vs Logger for log statements at a disabled level (ie, debug level when current log level is info).
* **FastLoggerEnabledLevelBenchmark** compares FastLogger vs Logger for log statements at an enabled level (ie, info level when current log level is info).
* **FastLoggerParameterTypeBenchmark** compares FastLogger vs Logger for various types of parameters at an enabled level (ie, info level when current log level is info).

Please review: @demery-pivotal @mhansonp @aaronlindsey @moleske @pivotal-jbarrett 

You can run a single JMH Benchmark class using:
```
$ ./gradlew geode-core:jmh -Pinclude=".*FastLoggerDisabledLevelBenchmark.*"
```
The numbers fluctuate from run-to-run, but _fastLogger_debugLogStatement_isDebugEnabled_benchmark_01_ consistently out-performs _logger_debugLogStatement_isDebugEnabled_benchmark_01_.

Sample run results:
```
Benchmark                                                                                   Mode  Cnt           Score   Error  Units

FastLoggerDisabledLevelBenchmark.fastLogger_debugLogStatement_isDebugEnabled_benchmark_01  thrpt       1807689366.083          ops/s
FastLoggerDisabledLevelBenchmark.logger_debugLogStatement_isDebugEnabled_benchmark_01      thrpt        619325332.223          ops/s
FastLoggerDisabledLevelBenchmark.fastLogger_debugLogStatement_benchmark_02                 thrpt        448305344.988          ops/s
FastLoggerDisabledLevelBenchmark.logger_debugLogStatement_benchmark_02                     thrpt        530355280.629          ops/s
FastLoggerDisabledLevelBenchmark.fastLogger_debugLogStatement_lambda_benchmark_03          thrpt        498633100.474          ops/s
FastLoggerDisabledLevelBenchmark.logger_debugLogStatement_lambda_benchmark_03              thrpt        627625938.074          ops/s
```
```
Benchmark                                                                         Mode  Cnt        Score   Error  Units

FastLoggerEnabledLevelBenchmark.fastLogger_infoLogStatement_benchmark_01         thrpt       3491924.597          ops/s
FastLoggerEnabledLevelBenchmark.logger_infoLogStatement_benchmark_01             thrpt       3407224.481          ops/s
FastLoggerEnabledLevelBenchmark.fastLogger_infoLogStatement_lambda_benchmark_02  thrpt       3057080.066          ops/s
FastLoggerEnabledLevelBenchmark.logger_infoLogStatement_lambda_benchmark_02      thrpt       3114615.700          ops/s
```
```
Benchmark                                                                                        Mode  Cnt        Score   Error  Units

FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_stringConcat_benchmark_01          thrpt       3110811.106          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_stringConcat_benchmark_01              thrpt       3020759.569          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_complexConcat_benchmark_02         thrpt         61516.196          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_complexConcat_benchmark_02             thrpt         61287.729          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_lambda_stringConcat_benchmark_03   thrpt       2710364.613          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_lambda_stringConcat_benchmark_03       thrpt       2667739.878          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_lambda_complexConcat_benchmark_04  thrpt         64283.010          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_lambda_complexConcat_benchmark_04      thrpt         60132.630          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_params_benchmark_05                thrpt       2944026.000          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_params_benchmark_05                    thrpt       2736174.880          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_complexParams_benchmark_06         thrpt       2926546.641          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_complexParams_benchmark_06             thrpt       2884928.362          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_lambda_params_benchmark_07         thrpt       2491212.879          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_lambda_params_benchmark_07             thrpt       2428997.756          ops/s
FastLoggerParameterTypeBenchmark.fastLogger_infoLogStatement_lambda_complexParams_benchmark_08  thrpt       2324345.545          ops/s
FastLoggerParameterTypeBenchmark.logger_infoLogStatement_lambda_complexParams_benchmark_08      thrpt       2275429.973          ops/s
```

### CODE REVIEW

**Comment by pivotal-jbarrett** [comment]
@moleske The benchmarks are more for being able to test changes you may make in this area or comparing new algorithms you come up with. There is little value trying to run them in the CI. The runtime of some of these can be in the hours too.

File: geode-core/src/jmh/java/org/apache/geode/internal/logging/log4j/FastLoggerDisabledLevelBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@Measurement(iterations = 1, time = 1, timeUnit = SECONDS)
```
Comment (pivotal-jbarrett) [inline geode-core/src/jmh/java/org/apache/geode/internal/logging/log4j/FastLoggerDisabledLevelBenchmark.java]: If your iterations and time are so low are you certain the JIT is coming into play on these benchmarks? Or is this just a hold over from you testing that this bench still compiles and runs?

Comment (kirklund) [reply geode-core/src/jmh/java/org/apache/geode/internal/logging/log4j/FastLoggerDisabledLevelBenchmark.java]: I set them small just for quick runs, but I do increase the values for various runs. The results of the longer runs are pretty much the same but if I run for 3+ hours then the results tend to be much more consistent than if you run several 1 second runs and compare them all. We can increase the configured values that are checked in if you prefer.

Comment (pivotal-jbarrett) [reply geode-core/src/jmh/java/org/apache/geode/internal/logging/log4j/FastLoggerDisabledLevelBenchmark.java]: Not necessary, just wanted to understand and make sure we evaluated longer runs.

**Pull request merged**